core/common/src/main/java/alluxio/exception/AlluxioException.java

```diff
@@ -76,15 +77,22 @@ public AlluxioTException toAlluxioTException() {
    * @return a {@link AlluxioException} of the type specified in e, with the message specified in e
    */
 
-  public static AlluxioException from(AlluxioTException e) {
+  /**
+   * Converts an Alluxio exception from Thrift representation to native representation.
+   *
+   * @param e the Alluxio Thrift exception
+   * @return the native Alluxio exception
+   */
+  public static AlluxioException fromThrift(AlluxioTException e) {
     try {
-      // For 1.0, the type returns the enum type, while for ALLUXIO 1.1 and after, the type contains
-      // the exception class name
-      Class<? extends AlluxioException> throwClassBackCompat =
-          AlluxioExceptionType.getAlluxioExceptionClass(e.getType());
-      @SuppressWarnings("unchecked")
-      Class<? extends AlluxioException> throwClass = throwClassBackCompat == null
-          ? (Class<? extends AlluxioException>) Class.forName(e.getType()) : throwClassBackCompat;
+      Class<? extends AlluxioException> throwClass;
+      if (e.isSetClassName()) {
+        // server version 1.1.0 or newer
+        throwClass = (Class<? extends AlluxioException>) Class.forName(e.getClassName());
+      } else {
+        // server version 1.0.x
+        throwClass = AlluxioExceptionType.getAlluxioExceptionClass(e.getType());
```

Comment:
If an exception is added in 1.8 and a 1.7 client receives it, it won't have the class definition on its classpath so it won't be able to instantiate it